File: ShadowFrameworkJS/src/shadow/geometry/SFGeometry.js

var SFGeometry_LOD_HINT_DISCARD=1;

function SFGeometry(){
	this.rendering_hint=0;
	this.available_lods=0;
	this.sonGeometries=new Array();
	this.changed=true;
}
	
		
SFGeometry.prototype["getRendering_hint"]=function(){
	return this.rendering_hint;
};


SFGeometry.prototype["getAvailable_lods"]=function(){
	return this.available_lods;
};
	
SFGeometry.prototype["setRendering_hint"]=function(rendering_hint){
	this.rendering_hint=rendering_hint;
};

SFGeometry.prototype["setAvailable_lods"]=function(available_lods){
	this.available_lods=available_lods;
};

SFGeometry.prototype["getSonsCount"]=function(){
	return this.sonGeometries.length;
};

SFGeometry.prototype["addSon"]=function(son){
	 this.sonGeometries.push(son);
	 this.son.setFatherGeometry(this);
	return this.sonGeometries.length-1;
};


SFGeometry.prototype["getSon"]=function(index){
	return this.sonGeometries[index];
};

SFGeometry.prototype["setSon"]=function(index){
	return this.sonGeometries[index];
};

SFGeometry.prototype["getFatherGeometry"]=function(){
	return this.fatherGeometry;
};

SFGeometry.prototype["setFatherGeometry"]=function(fatherGeometry){
	this.fatherGeometry=fatherGeometry;
};

SFGeometry.prototype["rebuild"]=function(){
	this.compile();
	for (var i = 0; i < this.sonGeometries.length; i++) {
		this.sonGeometries[i].rebuild();
	}
};
		

File: SFDataApplicationTestC++/shadow/geometry/SFGeometry.cpp
/*
 * SFGeometry.cpp
 *
 *  Created on: 24/set/2013
 *      Author: Alessandro
 */

#include "SFGeometry.h"

namespace sf {


SFGeometry::SFGeometry(){
	rendering_hint=0;
	available_lods=0;
	fatherGeometry=0;
	changed=true;
}

SFGeometry::~SFGeometry(){
	for(unsigned int i=0;i<sonGeometries.size();i++){
		delete sonGeometries[i];
	}
}


int SFGeometry::getRendering_hint() {
	return rendering_hint;
}

void SFGeometry::setRendering_hint(int rendering_hint) {
	this->rendering_hint = rendering_hint;
}

int SFGeometry::getAvailable_lods() {
	return available_lods;
}

void SFGeometry::setAvailable_lods(int available_lods) {
	this->available_lods = available_lods;
}

int SFGeometry::getSonsCount(){
	return sonGeometries.size();
}

int SFGeometry::addSon(SFGeometry* son){
	sonGeometries.push_back(son);
	son->setFatherGeometry(this);
	return sonGeometries.size()-1;
}

SFGeometry* SFGeometry::getSon(int index){
	return sonGeometries.at(index);
}

SFGeometry* SFGeometry::getFatherGeometry() {
	return fatherGeometry;
}

void SFGeometry::setFatherGeometry(SFGeometry* fatherGeometry) {
	this->fatherGeometry = fatherGeometry;
}


void SFGeometry::rebuild(){
	compile();
	for (unsigned int i = 0; i < sonGeometries.size(); i++) {
		sonGeometries[i]->rebuild();
	}
}

} /* namespace sf */
